processor/src/main/java/org/mapstruct/ap/model/common/Type.java

```diff
@@ -665,4 +690,57 @@ public int compareTo(Type o) {
     public String toString() {
         return typeMirror.toString();
     }
+
+
+    /**
+     *
+     * @return an identification that can be used as part in a forged method name.
+     */
+    public String getIdentification() {
+        if ( isArrayType() ) {
+            return componentType.getName() + "Array";
+        }
+        else {
+            return getTypeBound().getName();
+        }
+    }
+
+    /**
+     * Establishes the type bound:
+     * <ul>
+     * <li>{@code<? extends Number>}, returns Number</li>
+     * <li>{@code<? super Number>}, returns Number</li>
+     * <li>{@code<?>}, returns Object</li>
+     * <li>{@code<T extends Number>, returns Number</li>
+     * </ul>
+     * @return the bound for this parameter
+     */
+    public Type getTypeBound() {
+
+        Type boundingBase = this;
```

Comment:
yes it would. :smile:. I'll fix that. I initially thought of adding this in the `TypeFactory`. Anyway, I now fixed the `IterableMapping` and `MapMapping` but perhaps it should be added in the `PropertyMapping` as well (separate PR, don't want to make this too big).